core/src/main/java/org/hibernate/ogm/type/impl/TypeTranslatorImpl.java

```diff
@@ -121,4 +135,40 @@ else if ( type instanceof org.hibernate.type.CollectionType ) {
 		}
 		throw log.unableToFindGridType( type.getClass().getName() );
 	}
+
+	/*
+	   We directly pass the converter "database" type as value to the Tuple
```

Comment:
I'll open up a follow-up JIRA for this one.

Would we even need that registry you describe or couldn't we "just" take the target type from the converter (`UUID` in your example), find the right ORM `Type` for this and pass it to `GridDialect#overrideType()`? We could then feed the converted value through that grid type before writing it.